src/main/java/Duke.java

```diff
@@ -1,10 +1,169 @@
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+
 public class Duke {
+    final static String HORIZON = "____________________________________________________________\n";
+    static List<Task> tasks = new ArrayList<>();
+    static String withFormat(String str) {
+        return HORIZON + str + '\n' + HORIZON;
+    }
+
+    public static boolean isNumeric(String str) {
+        try {
+            Integer.parseInt(str);
+            return true;
+        } catch(NumberFormatException e){
+            return false;
+        }
+    }
+
+    static String validDelete(String input) {
+        if (!isNumeric(input.substring(7))) {
+            return "Pi-must be numbers behind-pi!";
+        } else if (Integer.parseInt(input.substring(7)) > tasks.size() || Integer.parseInt(input.substring(7)) <= 0) {
+            return "Pi-not within range-pi!";
+        } else {
+            int temp = Integer.parseInt(input.substring(7));
+            Task task = tasks.get(temp - 1);
+            tasks.remove(temp-1);
+            return "Pi-ka(Removed): " + task + '\n' +"Pikaaaaa: " + tasks.size() + (tasks.size() > 1 ? " tasks" : " task");
+        }
+    }
+
+    static String validMark(String input) {
+        if (!isNumeric(input.substring(5))) {
+            return "Pi-must be numbers behind-pi!";
+        } else if (Integer.parseInt(input.substring(5)) > tasks.size() || Integer.parseInt(input.substring(5)) <= 0) {
+            return "Pi-not within range-pi!";
```

Comment:
maybe these checks for valid input could be put into a separate method? since I noticed that the same checks are used in multiple other methods such as validDelete and validUnmark.

Reply:
That's true :( Will try to improve that in the future!